### PR TITLE
[main][feat] deploy script switch collateral router & dextors

### PR DESCRIPTION
**Pull request by Asaki-Katsu**

### CODE REVIEW

File: script/ts/commands/Dexter/curve-setPoolConfigs.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    config.extension.curveWstEthPool,
+    1,
+    0,
+    {
+      gasLimit: 10000000,
+    }
+  );
+  console.log(`[CurveDexter] Tx - Set Pool Config of (WSTETH, WETH): ${tx2.hash}`);
+  await tx2.wait(1);
+  console.log("[CurveDexter] Finished");
```
Comment (john-xina-p88) [inline script/ts/commands/Dexter/curve-setPoolConfigs.ts:35]: Make this a loop and let user config whatever we like

File: configs/arbitrum.goerli.json [hunk 2]
```diff
@@ -74,5 +76,18 @@
...
+    "dexter": {
+      "uniswapV3": "",
+      "curve": "",
+      "glp": ""
+    },
+    "uniswapConfig": {
+      "permit2": "0x000000000022d473030f116ddee9f6b43ac78ba3",
+      "universalRouter": "0x3fC91A3afd70395Cd496C647d5a6CC9D4B2b7FAD"
+    },
+    "curveWstEthPool": ""
```
Comment (john-xina-p88) [inline configs/arbitrum.goerli.json:91]: these should be in vendors
```
"vendors": {
    "gmx": {
      "glpManager": "0x3963FfC9dff443c2A94f21b129D429891E32ec18",
      "rewardRouterV2": "0xB95DB5B167D75e6d04227CfFFA61069348d271F5",
      "rewardTracker": "0x4e971a87900b931fF39d1Aad67697F49835400b6"
      "gmxVault": ""
    },
    "uniswap": {
      "permit2": "0x000000000022d473030f116ddee9f6b43ac78ba3",
      "universalRouter": "0x3fC91A3afd70395Cd496C647d5a6CC9D4B2b7FAD"
    },
    "curve": {
       "wstEthPool": ""
    }
  },
```

File: script/ts/commands/Dexter/uniswap-setPath.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { Command } from "commander";
```
Comment (john-xina-p88) [inline script/ts/commands/Dexter/uniswap-setPath.ts:1]: kebab case name pls

File: script/ts/commands/Dexter/uniswap-setPath.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  await tx1.wait(1);
+  const tx2 = await dexter.setPathOf(
+    config.tokens.weth,
+    config.tokens.arb,
+    ethers.utils.defaultAbiCoder.encode(["address", "uint24", "address"], [config.tokens.weth, 500, config.tokens.arb]),
+    {
+      gasLimit: 10000000,
+    }
+  );
+  console.log(`[UniswapDexter] Tx - Set Path of (WETH, ARB): ${tx2.hash}`);
```
Comment (john-xina-p88) [inline script/ts/commands/Dexter/uniswap-setPath.ts:30]: loop

File: script/ts/commands/SwitchCollateralRouter/setDexter.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import { Command } from "commander";
```
Comment (john-xina-p88) [inline script/ts/commands/SwitchCollateralRouter/setDexter.ts:1]: kebab case name

File: script/ts/commands/SwitchCollateralRouter/setDexter.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+import { loadConfig } from "../../utils/config";
+import signers from "../../entities/signers";
+import { SwitchCollateralRouter__factory } from "../../../../typechain";
+
+async function main(chainId: number) {
+  const config = loadConfig(chainId);
+  const deployer = signers.deployer(chainId);
+
+  const dexter = SwitchCollateralRouter__factory.connect(config.extension.switchCollateralRouter, deployer);
```
Comment (john-xina-p88) [inline script/ts/commands/SwitchCollateralRouter/setDexter.ts:10]: loop

File: script/ts/commands/Dexter/uniswap-set-path.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      tokenIn: config.tokens.weth,
+      tokenOut: config.tokens.arb,
+      path: ethers.utils.defaultAbiCoder.encode(
+        ["address", "uint24", "address"],
+        [config.tokens.weth, 500, config.tokens.arb]
+      ),
+    },
+  ];
+
+  console.log("[UniswapDexter] Setting path config...");
```
Comment (shutsuwei) [inline script/ts/commands/Dexter/uniswap-set-path.ts:37]: [cmds/UniswapDexter]

File: script/ts/commands/SwitchCollateralRouter/set-dexter.ts [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+      dexter: config.extension.dexter.curve,
+    },
+  ];
+
+  console.log("[SwitchCollateral] Setting dexter ...");
+  for (let i = 0; i < params.length; i++) {
+    const tx = await dexter.setDexterOf(params[i].tokenIn, params[i].tokenOut, params[i].dexter, {
+      gasLimit: 10000000,
+    });
+    console.log(`[SwitchCollateral] Tx - Set Dexter of (${params[i].tokenIn}, ${params[i].tokenOut}): ${tx.hash}`);
```
Comment (shutsuwei) [inline script/ts/commands/SwitchCollateralRouter/set-dexter.ts:54]: [cmds/SwitchCollateralRouter]

File: script/ts/deploys/Dexter/curve.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+import { getConfig, writeConfigFile } from "../../utils/config";
+
+const config = getConfig();
+
+async function main() {
+  const deployer = (await ethers.getSigners())[0];
+  const contract = await ethers.deployContract("CurveDexter", [config.tokens.weth], deployer);
+
+  await contract.deployed();
+  console.log(`Deploying CurveDexter Contract`);
```
Comment (shutsuwei) [inline script/ts/deploys/Dexter/curve.ts:11]: [deploys/Dexter]

File: script/ts/deploys/Dexter/glp.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+      config.tokens.sglp,
+      config.vendors.gmx.glpManager,
+      config.vendors.gmx.gmxVault,
+      config.vendors.gmx.rewardRouterV2,
+    ],
+    deployer
+  );
+
+  await contract.deployed();
+  console.log(`Deploying GlpDexter Contract`);
```
Comment (shutsuwei) [inline script/ts/deploys/Dexter/glp.ts:21]: [deploys/Dexter]

File: script/ts/deploys/Dexter/uniswapV3.ts [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+async function main() {
+  const deployer = (await ethers.getSigners())[0];
+  const contract = await ethers.deployContract(
+    "UniswapDexter",
+    [config.vendors.uniswap.permit2, config.vendors.uniswap.universalRouter],
+    deployer
+  );
+
+  await contract.deployed();
+  console.log(`Deploying UniswapDexter Contract`);
```
Comment (shutsuwei) [inline script/ts/deploys/Dexter/uniswapV3.ts:15]: [deploys/Dexter]

File: script/ts/deploys/SwitchCollateralRouter/index.ts [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+import { getConfig, writeConfigFile } from "../../utils/config";
+
+const config = getConfig();
+
+async function main() {
+  const deployer = (await ethers.getSigners())[0];
+  const contract = await ethers.deployContract("SwitchCollateralRouter", deployer);
+
+  await contract.deployed();
+  console.log(`Deploying SwitchCollateralRouter Contract`);
```
Comment (shutsuwei) [inline script/ts/deploys/SwitchCollateralRouter/index.ts:11]: [deploys/SwitchCollateralRouter]

**Review by shutsuwei — CHANGES_REQUESTED** [review]

**Review by john-xina-p88 — APPROVED** [review]

**Pull request closed without merging**